Nightly fails to compile core with Xargo on thumbv6m due to missing AtomicPtr

Here is the verbose output:

```

RUSTFLAGS="-C`` link-arg=-Tlayout_lpc1114.ld" xargo build --release --target thumbv6m-none-eabi --features "lpc1114" --verbose
+ "rustc" "--print" "sysroot"
+ "rustc" "--print" "target-list"
+ RUSTFLAGS="-C link-arg=-Tlayout_lpc1114.ld --sysroot /home/juergen/.xargo -Z force-unstable-if-unmarked"
+ "cargo" "build" "--release" "--manifest-path" "/tmp/xargo.jJOPpAK3kHEu/Cargo.toml" "--target" "thumbv6m-none-eabi" "-v" "-p" "core"
   Compiling core v0.0.0 (file:///home/juergen/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore)
     Running `rustc --crate-name core /home/juergen/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=30eb9fdc2904bb34 -C extra-filename=-30eb9fdc2904bb34 --out-dir /tmp/xargo.jJOPpAK3kHEu/target/thumbv6m-none-eabi/release/deps --target thumbv6m-none-eabi -L dependency=/tmp/xargo.jJOPpAK3kHEu/target/thumbv6m-none-eabi/release/deps -L dependency=/tmp/xargo.jJOPpAK3kHEu/target/release/deps -C link-arg=-Tlayout_lpc1114.ld --sysroot /home/juergen/.xargo -Z force-unstable-if-unmarked`
error[E0412]: cannot find type `AtomicPtr` in this scope
   --> /home/juergen/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/sync/atomic.rs:931:26
    |
931 | impl<T> From<*mut T> for AtomicPtr<T> {
    |                          ^^^^^^^^^ not found in this scope

error: aborting due to previous error

error: Could not compile `core`.

Caused by:
  process didn't exit successfully: `rustc --crate-name core /home/juergen/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=30eb9fdc2904bb34 -C extra-filename=-30eb9fdc2904bb34 --out-dir /tmp/xargo.jJOPpAK3kHEu/target/thumbv6m-none-eabi/release/deps --target thumbv6m-none-eabi -L dependency=/tmp/xargo.jJOPpAK3kHEu/target/thumbv6m-none-eabi/release/deps -L dependency=/tmp/xargo.jJOPpAK3kHEu/target/release/deps -C link-arg=-Tlayout_lpc1114.ld --sysroot /home/juergen/.xargo -Z force-unstable-if-unmarked` (exit code: 101)
error: `"cargo" "build" "--release" "--manifest-path" "/tmp/xargo.jJOPpAK3kHEu/Cargo.toml" "--target" "thumbv6m-none-eabi" "-v" "-p" "core"` failed with exit code: Some(101)

```

--- a/src/libcore/sync/atomic.rs
+++ b/src/libcore/sync/atomic.rs
@@ -927,6 +927,7 @@ impl<T> AtomicPtr<T> {
     }
 }
 
+#[cfg(target_has_atomic = "ptr")]
 #[stable(feature = "atomic_from", since = "1.23.0")]
 impl<T> From<*mut T> for AtomicPtr<T> {
     #[inline]

